File: backend/src/main/java/com/e108/be/domain/record/dto/response/CalendarResponse.java
package com.e108.be.domain.record.dto.response;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Builder
public class CalendarResponse {

    private int year;
    private int month;
    private CalendarSummary summary;
    private List<DayWalkCount> days;

    @Getter
    @Builder
    public static class CalendarSummary {
        private int walkRate;
        private int totalDurationMinutes;
        private double totalDistanceKm;
    }

    @Getter
    @Builder
    public static class DayWalkCount {
        private LocalDate date;
        private int walkCount;
    }
}

File: frontend/app/src/main/java/com/frontend/domain/model/CalendarResponse.kt
package com.frontend.domain.model

data class CalendarResponse(
    val year: Int,
    val month: Int,
    val summary: CalendarSummary,
    val days: List<DayWalkCount>
)

data class CalendarSummary(
    val walkRate: Int,
    val totalDurationMinutes: Int,
    val totalDistanceKm: Double
)

data class DayWalkCount(
    val date: String,   // "2026-03-12"
    val walkCount: Int
)
